src/test/java/uk/gov/justice/digital/delius/CourtReportAPITest.java

```diff
@@ -0,0 +1,326 @@
+import uk.gov.justice.digital.delius.jwt.Jwt;
+import uk.gov.justice.digital.delius.service.CourtReportService;
+import uk.gov.justice.digital.delius.service.OffenderService;
+import uk.gov.justice.digital.delius.user.UserData;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@RunWith(SpringJUnit4ClassRunner.class)
+@DirtiesContext
+public class CourtReportAPITest {
+
+    @LocalServerPort
+    int port;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private Jwt jwt;
+
+    @MockBean
+    private OffenderService offenderService;
+
+    @MockBean
+    private CourtReportService courtReportService;
+
+    @MockBean
+    private CourtReportRepository courtReportRepository;
+
+
+
+    @Before
+    public void setup() {
+        RestAssured.port = port;
+        RestAssured.basePath = "/api";
+        RestAssured.config = RestAssuredConfig.config().objectMapperConfig(new ObjectMapperConfig().jackson2ObjectMapperFactory(
+                (aClass, s) -> objectMapper
+        ));
+        when(offenderService.offenderIdOfCrn("CRN1")).thenReturn(Optional.of(1L));
+        when(offenderService.offenderIdOfNomsNumber("NOMS1")).thenReturn(Optional.of(1L));
+        when(courtReportService.courtReportsFor(any())).thenReturn(ImmutableList.of(
+            CourtReport.builder().courtReportId(1L).offenderId(1L).dateRequested(LocalDateTime.now()).build(),
+            CourtReport.builder().courtReportId(2L).offenderId(1L).dateRequested(LocalDateTime.now()).build(),
+            CourtReport.builder().courtReportId(4L).offenderId(1L).dateRequested(LocalDateTime.now()).build()
+        ));
+        when(courtReportService.courtReportFor(any(), any())).thenReturn(
+                Optional.of(CourtReport.builder().courtReportId(4L).offenderId(1L).build()));
+    }
+
+    @Test
+    public void canGetAllReportsForOffenderByCrn() {
+
+        CourtReport[] courtReports = given()
+            .header("Authorization", aValidToken())
+            .when()
+            .get("offenders/crn/CRN1/courtReports")
+            .then()
+            .statusCode(200)
+            .extract()
+            .body()
+            .as(CourtReport[].class);
+
+        assertThat(courtReports).hasSize(3);
+    }
+
+    @Test
+    public void missingOffenderRecordResultsInAllReportsForOffenderByCrnNotFound() {
+        when(offenderService.offenderIdOfCrn("CRN1")).thenReturn(Optional.empty());
+
+        given()
+            .header("Authorization", aValidToken())
+            .when()
+            .get("offenders/crn/CRN1/courtReports")
+            .then()
+            .statusCode(404);
+
```

Comment:
Nick, did you put a check in here to ensure that we don't get a false positive for 404